cuda_core/cuda/core/experimental/_dlpack.pyx

```diff
@@ -33,6 +34,16 @@ cdef void deleter(DLManagedTensor* tensor) noexcept with gil:
     stdlib.free(tensor)
 
 
+cdef void cleanup(DLManagedTensor* tensor) noexcept with gil:
```

Comment:
Q: Can we merge this with `deleter` above?

Reply:
As we're delaying the user-facing dlpack export for now, I reverted dlpack changes here not to inflate the PR or introduce unused code-paths.

I think the dlpack capsule creation needs to be fixed not to leak stuff if exception happens during its creation. I'll open another small PR for that.